app/src/main/java/com/futurice/freesound/viewmodel/DisplayableItem.java

```diff
@@ -0,0 +1,46 @@
+package com.futurice.freesound.viewmodel;
+
+import com.google.auto.value.AutoValue;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Wraps a model. Convenient for complex presentation layers such as
+ * {@link android.support.v7.widget.RecyclerView} where different types of model are handled.
+ */
+@AutoValue
+public abstract class DisplayableItem<T> {
+
+    // Types
+    public static final int SOUND = 0;
```

Comment:
I first did it with enums but I didn't like the solution and decided to go for the simplicity of the int constants instead.